contrib/test.sh

```diff
@@ -2,6 +2,14 @@
 
 FEATURES="base64 bitcoinconsensus use-serde rand"
 
+# Old cargo format is incompatible with a newer ones, so we need to clean this up
+rm -f Cargo.lock
+# Use toolchain if explicitly specified
+if [ -n "$TOOLCHAIN" ]
+then
+    alias cargo='cargo +"${TOOLCHAIN}"'
```

Comment:
I wonder if we should replace `PIN_VERSIONS` with a check for the toolchain version? I removed that some time ago because the toolchain variable was used for nothing else, but now that might make sense again …

Reply:
This will break the script once we update MSRV soon (and of course we will forget to update it here). Also script may be used for checking with custom toolchains and we cant do a reliable detection when we need to pin the versions

Reply:
> This will break the script once we update MSRV

It should, we should unpin as many deps as possible then. But yeah, when you don't want to set the toolchain that way it's still useful to have it as a separate flag.

Reply:
I occasionally run tests on 1.36 w/o pins and it failed, so pinning probably will not change...